query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -63,10 +66,12 @@
     // allSelections contain all the various expressions in QueryRequest that we want selections on.
     // Group bys, selections and aggregations in that order. See RequestAnalyzer#analyze() to see
     // how it is created.
+    int selectionCounter=0;
```

Comment:
What's a selection counter and why do we need it?

Reply:
There can be multiple selections in a query request. So selectionCounter is basically an index which tells that this selection appeared at this index/position in the request.

This is needed because we need to remember which selection appeared at which index. (needed for AVG_RATE implementation)